mantle/kola/tests/misc/boot-mirror-luks.go

```diff
@@ -0,0 +1,273 @@
+				},
+				{
+				  "label": "esp-2",
+				  "sizeMiB": 127,
+				  "typeGuid": "C12A7328-F81F-11D2-BA4B-00A0C93EC93B"
+				},
+				{
+				  "label": "boot-2",
+				  "sizeMiB": 384
+				},
+				{
+				  "label": "root-2"
+				}
+			  ],
+			  "wipeTable": true
+			}
+		  ],
+		  "filesystems": [
+			{
+			  "device": "/dev/disk/by-partlabel/esp-1",
+			  "format": "vfat",
+			  "label": "esp1",
+			  "wipeFilesystem": true
+			},
+			{
+				"device": "/dev/disk/by-partlabel/esp-2",
+				"format": "vfat",
+				"label": "esp2",
+				"wipeFilesystem": true
+			},
+			{
+			  "device": "/dev/md/md-boot",
+			  "format": "ext4",
+			  "label": "boot",
+			  "wipeFilesystem": true
+			},
+			{
+			  "device": "/dev/mapper/root",
+			  "format": "xfs",
+			  "label": "root",
+			  "wipeFilesystem": true
+			}
+		  ],
+		  "luks": [
+			{
+			  "clevis": {
+				"tpm2": true
+			  },
+			  "device": "/dev/md/md-root",
+			  "label": "luks-root",
+			  "name": "root",
+			  "wipeVolume": true
+			}
+		  ],
+		  "raid": [
+			{
+			  "devices": [
+				"/dev/disk/by-partlabel/boot-1",
+				"/dev/disk/by-partlabel/boot-2"
+			  ],
+			  "level": "raid1",
+			  "name": "md-boot",
+			  "options": [
+				"--metadata=1.0"
+			  ]
+			},
+			{
+			  "devices": [
+				"/dev/disk/by-partlabel/root-1",
+				"/dev/disk/by-partlabel/root-2"
+			  ],
+			  "level": "raid1",
+			  "name": "md-root"
+			}
+		  ]
+		}
+	  }`)
+)
+
+func mustMatch(c cluster.TestCluster, r string, output []byte) {
+	m, err := regexp.Match(r, output)
+	if err != nil {
+		c.Fatalf("Failed to match regexp %s: %v", r, err)
+	}
+	if !m {
+		c.Fatalf("Regexp %s did not match text: %s", r, output)
+	}
+}
+
+func mustNotMatch(c cluster.TestCluster, r string, output []byte) {
+	m, err := regexp.Match(r, output)
+	if err != nil {
+		c.Fatalf("Failed to match regexp %s: %v", r, err)
+	}
+	if m {
+		c.Fatalf("Regexp %s matched text: %s", r, output)
+	}
+}
+
+func init() {
+	register.RegisterTest(&register.Test{
+		Run:                  runBootMirrorLUKSTest,
+		ClusterSize:          0,
+		Name:                 `coreos.boot-mirror.luks`,
+		Platforms:            []string{"qemu-unpriv"},
+		ExcludeArchitectures: []string{"s390x"}, // no TPM support for s390x in qemu
+		Tags:                 []string{"boot-mirror", "luks", "raid1", "tpm2", kola.NeedsInternetTag},
+	})
+}
+
+// runBootMirrorLUKSTest verifies if the boot-mirror+LUKS RAID1
+// flow works properly in both BIOS and UEFI modes.
+func runBootMirrorLUKSTest(c cluster.TestCluster) {
+	var m platform.Machine
+	var err error
+	options := platform.QemuMachineOptions{
+		MachineOptions: platform.MachineOptions{
+			AdditionalDisks: []string{"5120M"},
+			MinMemory:       4096,
+		},
+	}
+	m, err = c.Cluster.(*unprivqemu.Cluster).NewMachineWithQemuOptions(bootmirrorluks, options)
+	if err != nil {
+		c.Fatal(err)
+	}
+	luksSanityTest(c, m, true)
+	// Check for root
+	c.MustSSH(m, "sudo mdadm -D /dev/md/md-root")
```

Comment:
WDYT about checking what devices are apart of both `/dev/md/md-root` & `/dev/md/md-boot` instead of just doing a single call to `mdadm` querying one of the raids without checking the output?